packages/angular_devkit/build_angular/src/browser/index.ts

```diff
@@ -592,34 +595,55 @@ export function buildWebpackBrowser(
               }
 
               let bundleInfoText = '';
-              const processedNames = new Set<string>();
-              for (const result of processResults) {
-                processedNames.add(result.name);
-
-                const chunk =
-                  webpackStats &&
-                  webpackStats.chunks &&
-                  webpackStats.chunks.find(c => result.name === c.id.toString());
-                if (result.original) {
-                  bundleInfoText +=
-                    '\n' + generateBundleInfoStats(result.name, result.original, chunk);
-                }
-                if (result.downlevel) {
-                  bundleInfoText +=
-                    '\n' + generateBundleInfoStats(result.name, result.downlevel, chunk);
-                }
-              }
-
               if (webpackStats && webpackStats.chunks) {
                 for (const chunk of webpackStats.chunks) {
-                  if (processedNames.has(chunk.id.toString())) {
-                    continue;
+                  const result = processResults.find((result) => result.name === chunk.id.toString());
+                  if (result) {
+                    if (result.original) {
+                      bundleInfoText +=
+                        '\n' + generateBundleInfoStats(result.name, result.original, chunk);
+                    }
+                    if (result.downlevel) {
+                      bundleInfoText +=
+                        '\n' + generateBundleInfoStats(result.name, result.downlevel, chunk);
+                    }
+                  } else {
+                    const asset =
+                    webpackStats.assets && webpackStats.assets.find(a => a.name === chunk.files[0]);
+                    bundleInfoText +=
+                      '\n' + generateBundleStats({ ...chunk, size: asset && asset.size }, true);
                   }
+                }
+              }
 
-                  const asset =
-                    webpackStats.assets && webpackStats.assets.find(a => a.name === chunk.files[0]);
-                  bundleInfoText +=
-                    '\n' + generateBundleStats({ ...chunk, size: asset && asset.size }, true);
+              const budgets = options.budgets || [];
+              for (const budget of budgets) {
+                const sizes = calculateSizes(budget, webpackStats);
+                for (const threshold of calculateThresholds(budget)) {
+                  for (const {size, label} of sizes) {
+                    if (exceededThreshold(size, threshold)) {
+                      const colorize = colorizeSeverity(threshold.severity);
+                      const severity = stringifySeverity(threshold.severity);
```

Comment:
`ThresholdSeverity` is a string enum so `toUpperCase` can be called directly on `threshold.severity`.

Reply:
This function is more about decoupling than anything else. If the format of `ThresholdSeverity` changes (say to `'W'` and `'E'`), then we would not want this text to change. So this function is just adding a layer of indirection between the view and model.

Reply:
`stringifySeverity()` got deleted anyways since we're now using `compilation.warnings` and `compilation.errors` to render them to the user, so nothing to do here.